doc/developer/design/20220516_transactional_consistency.md

```diff
@@ -0,0 +1,456 @@
+# Materialize Transactional Consistency
+
+## Summary
+
+A description of Materialize’s consistency guarantees with respect to the operations that occur inside of Materialize.
+Operations include:
+
+- `SELECT`, `INSERT`, `UPDATE`, and `DELETE` statements (but not `TAIL`)
+
+This design document is in service of [Epic #11631](https://github.com/MaterializeInc/materialize/issues/11631)
+and [Correctness Property #1](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/platform/ux.md#correctness)
+. Specifically for the first bullet point from the correctness property, “`SELECT`, `INSERT`, `UPDATE`, and `DELETE`
+statements (but not `TAIL`)". A follow-up design document will be created for the second bullet point, "Materialize
+initiated acknowledgements of upstream sources (e.g., the commit of an offset by a Kafka source, the acknowledge of an
+LSN by a PostgresSQL source, etcetera)".
+
+This document does not take any opinion on whether the changes proposed should be the default behavior or an opt-in
+behavior via some configuration.
+
+## Goals
+
+The goal of this design document is to describe the transactional consistency guarantees that are provided by
+Materialize and describe what changes are needed to provide higher levels of transaction consistency. This document will
+focus on Read Uncommitted, Read Committed, Repeatable Read, Serializable, and Strict Serializable consistency levels.
+
+## Non-Goals
```

Comment:
tw nit: to make things scannable, think in terms of lists and tables, i.e. can you add structure to the doc to replace words

Reply:
I don't fully understand this comment, what do you mean by that?

Reply:
If you think in terms of trying to structure the content, rather than "writing it," you could reduce this to something like:

```
The following items' impact on consistency is not considered:
- `TAIL`
- Sinks
- `AS OF`
```